src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -144,6 +145,38 @@ public int numOfSessionBetween(SessionDate inclusiveStart, SessionDate inclusive
         return (firstInSpan.numOfDayTo(lastInSpan) / interval.getValue()) + 1;
     }
 
+    /**
+     * Checks if the {@code Session} slot overlaps with all recurring sessions.
+     * @param otherSession the other session that is compared to.
+     */
+    public boolean isOverlapping(Session otherSession) {
+        return otherSession.isOverlapping(this);
+    }
+
+    /**
+     * Checks if all {@code RecurringSession} slots overlaps with all recurring sessions.
+     * @param otherSession the other recurring session that is compared to.
+     */
+    public boolean isOverlapping(RecurringSession otherSession) {
+        if (startAfter(otherSession.getSessionDate())) {
+            return otherSession.isOverlapping(this);
+        }
+        int daysBetween = (int) ChronoUnit.DAYS
+                .between(getSessionDate().getDate(), otherSession.getSessionDate().getDate());
+
+        int intervalMod = getInterval().getValue() % otherSession.getInterval().getValue();
+        int startDiffMod = daysBetween % otherSession.getInterval().getValue();
+        int diffMod = intervalMod % startDiffMod;
+        if (diffMod != 0) {
+            return false;
```

Comment:
try 
daysBetween = 8,
interval1 = 3 and interval2 =7 and vice versa.
They should both be true, but this gives false in former case

Reply:
gcd(interval1, interval2) |  daysBetween <--> there will be an overlap at some point

this is because there exists some overlap if n only if
`daysBetween + (interval1 x n) = interval2 x m`

The diophantine equation ax+by=c, has solutions if and only if gcd(a,b)|c.
gcd(interva1, interval2) | daysBetween, means daysBetween is divisible by gcd(i1, i2)

Reply:
So i think this should make sense :

     public boolean isOverlapping(RecurringSession otherSession) {
        if (startAfter(otherSession.getSessionDate())) {
            return otherSession.isOverlapping(this);
        }
        int daysBetween = getSessionDate.daysTo(otherSession.getSessionDate);
        
        if (daysBetween % gcd(interval1, interval2) !=0) {
            return false;
        }
... followed by time check